### ISSUE TITLE
snowflake_storage_integration fail on subsequent runs

### ISSUE THREAD

**Issue by hendrikhalkow** [chanzuckerberg/terraform-provider-snowflake#325]
**Provider Version**

0.18.2

**Terraform Version**

0.14.0

**Describe the bug**

Creating the following resource works fine:

```hcl
resource "snowflake_storage_integration" "s3" {
  name                 = "S3_${upper(var.environment)}"
  type                 = "EXTERNAL_STAGE"
  storage_provider     = "S3"
  enabled              = true
  storage_aws_role_arn = "arn:aws:iam::${var.aws_account_id}:role/snowflake"
  storage_allowed_locations = [
    "s3://${aws_s3_bucket.snowflake.bucket}/",
  ]
}
```

However, on subsequent Terraform runs, there is always a change to be applied, which is the type attribute. Applying that change fails. When I try to implement a workaround by adding lifecycle ignore changes for the type attibute, I get this:

```
expected type to be one of [EXTERNAL_STAGE], got 
```

**Expected behavior**

All subsequent terraform runs complete successfully and the lifecycle ignore changes workaround is not needed.

**Code samples and commands**

see above.

**Additional context**

Happens on both local system and Terraform Cloud

**Comment by aidan-melen** [comment]
Workaround: https://github.com/chanzuckerberg/terraform-provider-snowflake/issues/204#issuecomment-736865140

**Comment by hendrikhalkow** [comment]
See above. The workaround with lifecycle ignore_changes does not work.

**Comment by aidan-melen** [comment]
What happens when you run?

```hcl
resource "snowflake_storage_integration" "s3" {
  name                 = "S3_${upper(var.environment)}"
  # type                 = "EXTERNAL_STAGE"
  storage_provider     = "S3"
  enabled              = true
  storage_aws_role_arn = "arn:aws:iam::${var.aws_account_id}:role/snowflake"
  storage_allowed_locations = [
    "s3://${aws_s3_bucket.snowflake.bucket}/",
  ]

  # https://github.com/chanzuckerberg/terraform-provider-snowflake/issues/204
  lifecycle {
    ignore_changes = [type]
  }
}
```

**Comment by hendrikhalkow** [comment]
The first time, everything works fine. When I run `terraform plan` the next time, it fails with the following error message.

```
Error: expected type to be one of [EXTERNAL_STAGE], got 
```

**Comment by ryanking** [comment]
This might be fixed in #346 if you want to try that branch.

**Comment by ryanking** [comment]
Going to close this hoping that #346 fixes, please re-open if it does not
